scripts/teamDataScript.js

```diff
@@ -0,0 +1,617 @@
+        {
+            var pair = pitObjValue[k];
+            var key = Object.keys(pair)[0];
+            var answer = pair[key];
+            newRowHTML += `<td>${answer}</td>`;
+        }
+        //newRowHTML += '</tr>' //end of row
+
+        $('#allTeamsPitAnswers').append(newRowHTML);
+    }
+}
+
+
+
+//Match Stuff
+function loadMatch()
+{
+    $('#currMatchHeader').text('');
+    var season = getCookie('currCheckedSeason');
+    var event = getCurrEvent();
+    var teamNum = getCurrTeamNumber();
+
+    if (teamNum.trim().length == 0)
+    {
+        $('#matchTableConfirmation').text('No Team selected.');
+        $('#matchNumberStuff').hide();
+        return;
+    }
+    $('#matchNumberStuff').show();
+    loadMatchNumberList();
+
+    var tableHeader = '<tr> <th><b>Element</b></th> <th><b>Frequency</b></th> </tr>';
+    $('#matchQuestionTable').html(tableHeader); //default start of table
+
+    //retrieve elements for current season
+    var cname = `/season_config/${season}/` //starter bit of season_config_name
+    var cookieList = document.cookie.split(';');
+
+    for (var i = 0; i < cookieList.length; i++)
+    {
+        var currCName = cookieList[i].trim().split('=')[0];
+
+        if (!(currCName.startsWith(cname))) continue;
+
+        var element = currCName.substring(currCName.lastIndexOf('/')+1);
+        var value = getCookie(currCName);
+
+        var inputHTML = `<input type="text" class="q-input">`;
+        var elementText = `${element}`;
+        var newTRHtml = '<tr><td>' + elementText + '</td><td>' + inputHTML + "</td></tr>";
+        
+        if ($('#matchQuestionTable').html().indexOf(elementText) < 0) //if table doesn't already have it
+        {
+            $('#matchQuestionTable').append(newTRHtml);
+        }
+    }
+}
+
+function addNewMatch() //onclick for adding the button to add a match number
+{
+    var season = getCookie('currCheckedSeason');
+    var event = getCurrEvent();
+    var team = getCurrTeamNumber();
+    var newMatchNum = $('#matchNumberInputBox').val().trim();
+
+    //store match numbers in a local object: {'/season/event/team/match numbers' : " '1','7','20'... "}
+    var objName = matchNumberObjName(season,event,team);
+
+    //check if local storage object already has the new match number, stores an array of match numbers
+    var storedString = localStorage.getItem(objName);
+    var alreadyStored;
+    var numsArr = [];
+    if (!checkForNull(storedString) || storedString.length == 0) 
+    {
+        storedString = '';
+        alreadyStored = false;
+    } else
+    {
+        numsArr = storedString.split(','); //arr.toString() values seperated by ,
+        for (var i = 0; i < numsArr.length; i++)
+        {
+            var currNum = numsArr[i];
+            if (parseInt(currNum) == parseInt(newMatchNum))
+            {
+                alreadyStored = true;
+                break;
```

Comment:
breaking out of loops is considered bad. Sometimes you can't avoid it, but in this case, change the loop condition to `i < numsArr.length && !alreadyStored` and remove this break so that when you first come to the loop you "see" that it is looping through the array till `alreadyStored` is determined, and the contents of the loop are only focused on how to determine that (and not on breaking out).